components/dashboard/dashboard.go

```diff
@@ -0,0 +1,35 @@
+package dashboard
+
+import (
+	dsc "github.com/opendatahub-io/opendatahub-operator/apis/datasciencecluster/v1alpha1"
+	"github.com/opendatahub-io/opendatahub-operator/components"
+	"github.com/opendatahub-io/opendatahub-operator/components/profiles"
+	"github.com/opendatahub-io/opendatahub-operator/pkg/deploy"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	ComponentName = "odh-dashboard"
+	Path          = "/opt/odh-manifests/odh-dashboard/base"
+)
+
+type Dashboard struct {
+	components.Component `json:""`
+}
+
+// Verifies that Dashboard implements ComponentInterface
+var _ components.ComponentInterface = (*Dashboard)(nil)
+
+func (d *Dashboard) ReconcileComponent(instance *dsc.DataScienceCluster, client client.Client, scheme *runtime.Scheme, plan *profiles.ReconciliationPlan, namespace string) error {
+
+	if plan.Components[ComponentName] {
```

Comment:
I think this 'if' needs to be removed. For the case where the component was enabled and then subsequently disabled, we need to call DeployManifestFromPath so that the reconciliation logic can delete previously created resources.